mars/core/graph/entity.py

```diff
@@ -74,6 +84,20 @@ def results(self):
     def results(self, new_results):
         self._result_tileables = new_results
 
+    @property
+    def logic_key(self):
+        if not hasattr(self, "_logic_key") or self._logic_key is None:
+            token_keys = []
+            for node in self.topological_iter():
+                token_keys.append(
+                    tokenize(node.op.get_logic_key(), **node.extra_params)
```

Comment:
As keys already determinded for nodes, is tehre any extra information brought by extra_params? Remember that `tokenize` is an expensive call.

Reply:
Thanks for your reminder. There are `raw_chunk_size` and other info in extra_params which may make the chunk graph is different. This logic key is used when communicating with external service and its cost is comparable to the time consumed by requesting external service.

Reply:
I would like `tokenize` not performend when there is no `extra_params`.

Reply:
Good suggestion.